### ISSUE TITLE
Google Cloud Gateway module

### ISSUE THREAD

**Issue by jamilbk** [CloudFire-LLC/cloudfire#6044]
Move Google gateway module to dedicated repo and versioned like the Azure/AWS gateway module.

**Comment by AndrewDryga** [comment]
What would be less opinionated? Right now it just creates a deployment using region instance manager.

**Comment by jamilbk** [comment]
Do customers need all these services to run a Firezone Gateway in GCP?

https://github.com/firezone/firezone/blob/main/terraform/modules/google-cloud/apps/gateway-region-instance-group/services.tf

**Comment by AndrewDryga** [comment]
Yes, they are generally required to run anything on Google Compute, and they just make sure that APIs are enabled (which should be true for default project) - you don't have to use them.

**Comment by jamilbk** [comment]
github.com/firezone/terraform-google-gateway
